manifests/fact.pp

```diff
@@ -6,7 +6,7 @@
   $value,
   String $fact                    = $name,
   String $file                    = 'facts.txt',
-  Stdlib::Absolutepath $facts_dir = '/etc/facter/facts.d',
+  Stdlib::Absolutepath $facts_dir = $::facter::facts_d_dir,
```

Comment:
I didnt think lookups like this worked, since the facter class may not yet be in the catalog.

Reply:
I think that's why you need the precondition. The precondition should not be needed, because the define will include the facter class.

Reply:
Yeh, as you say, if `facter` class isn't already included, then this won't work... 

However this is a model I've seen used in several places, including `puppet-nginx` - https://github.com/voxpupuli/puppet-nginx/blob/master/manifests/resource/server.pp#L185
I think the convenience of not having to specify `facts_dir` on every `facter::fact` call outweighs the need to do an `include ::facter` somewhere...